src/share/vm/c1/c1_LIRGenerator.cpp

```diff
@@ -454,13 +454,11 @@ CodeEmitInfo* LIRGenerator::state_for(Instruction* x, ValueStack* state, bool ig
 
     MethodLivenessResult liveness = method->liveness_at_bci(bci);
     if (bci == SynchronizationEntryBCI) {
-      if (x->as_ExceptionObject() || x->as_Throw()) {
```

Comment:
please add comment here to make the change here clear

Reply:
done.